compat/mingw.c

```diff
@@ -1498,10 +1499,40 @@ struct pinfo_t {
 static struct pinfo_t *pinfo = NULL;
 CRITICAL_SECTION pinfo_cs;
 
+#ifndef SIGRTMAX
+#define SIGRTMAX 63
+#endif
+
+static void kill_child_processes_on_signal(void)
+{
+	DWORD status;
+
+	/*
+	 * Only continue if the process was terminated by a signal, as
+	 * indicated by the exit status (128 + sig_no).
+	 */
+	if (!GetExitCodeProcess(GetCurrentProcess(), &status) ||
```

Comment:
No, as the atexit handler is a little bit special: the exit code has been set at this point.